fmralign/template_alignment.py

```diff
@@ -0,0 +1,353 @@
+        because noise will cancel across images
+
+    Returns
+    -------
+    average_img: Niimg
+        Average of imgs, with same shape as one img
+    """
+    masked_imgs = [masker.transform(img) for img in imgs]
+    average_img = np.mean(masked_imgs, axis=0)
+    scale = 1
+    if scale_template:
+        X_norm = 0
+        for img in masked_imgs:
+            X_norm += np.linalg.norm(img)
+        X_norm /= len(masked_imgs)
+        scale = X_norm / np.linalg.norm(average_img)
+    average_img *= scale
+
+    return masker.inverse_transform(average_img)
+
+
+def _align_images_to_template(imgs, template, alignment_method,
+                              n_pieces, clustering_method, n_bags, masker,
+                              memory, memory_level, n_jobs, parallel_backend,
+                              verbose):
+    '''Convenience function : for a list of images, return the list
+    of estimators (PairwiseAlignment instances) aligning each of them to a
+    common target, the template. All arguments are used in PairwiseAlignment
+    '''
+    aligned_imgs = []
+    for img in imgs:
+        piecewise_estimator = \
+            PairwiseAlignment(n_pieces=n_pieces,
+                              alignment_method=alignment_method,
+                              clustering_method=clustering_method, n_bags=n_bags,
+                              mask=masker, memory=memory,
+                              memory_level=memory_level,
+                              n_jobs=n_jobs, parallel_backend=parallel_backend,
+                              verbose=verbose)
+        piecewise_estimator.fit(img, template)
+        aligned_imgs.append(piecewise_estimator.transform(img))
+    return aligned_imgs
+
+
+def _create_template(imgs, n_iter, scale_template, alignment_method, n_pieces,
+                     clustering_method, n_bags, masker, memory, memory_level,
+                     n_jobs, parallel_backend, verbose):
+    '''Create template through alternate minimization.  Compute iteratively :
+        * T minimizing sum(||R_i X_i-T||) which is the mean of aligned images (RX_i)
+        * align initial images to new template T
+            (find transform R_i minimizing ||R_i X_i-T|| for each img X_i)
+
+
+        Parameters
+        ----------
+        imgs: List of Niimg-like objects
+           See http://nilearn.github.io/manipulating_images/input_output.html
+           source data. Every img must have the same length (number of sample)
+        scale_template: boolean
+            rescale template after each inference so that it keeps
+            the same norm as the average of training images.
+        n_iter: int
+           number of iterations in the alternate minimization. Each img is
+           aligned n_iter times to the evolving template. If n_iter = 0,
+           the template is simply the mean of the input images.
+        All other arguments are the same are passed to PairwiseAlignment
+    '''
```

Comment:
Why not the return arguments ?

Reply:
I'm not sure I got your question right.

Reply:
You describe the parameters not the return arguments.